### ISSUE TITLE
Vanilla FL: Adam optimizer doesn't work but SGD does, problem with add_()

### ISSUE THREAD

**Issue by NiWaRe** [OpenMined/PySyft#3349]
**Description**
Based on the issue #3338 I ran a Vanilla FL process on a small ResNet on the Cifar10 dataset. 
I believe #3338 should be resolved, but I noticed an error which I can't understand. 
I ran the below code a couple of times with SGD optimizer and no lr-scheduler (and no momentum) which resulted in a perfectly normal training process (after 6 epochs around 60% acc. for plain ResNet18) 
Running the code multiple times with standard Adam optimizer however the following error produced itself at 50% through the first epoch. I was under the impression (as also shown in #3293 and in [this tutorial](https://github.com/OpenMined/PySyft/blob/master/examples/tutorials/Part%2007%20-%20Federated%20Learning%20with%20Federated%20Dataset.ipynb) ) that moment-based optimizers can be used. (Especially in non Vanilla FL without SMPC why shouldn't it?)

**Error-Message:** 
In Short: To expand please visit the Google Colab Link
<img width="1348" alt="Screenshot 2020-04-13 11 20 52" src="https://user-images.githubusercontent.com/13786323/79109570-e59c6180-7d78-11ea-94c6-b81d24bb477a.png">

**To Reproduce**
The following code (altered from #3293) was executed with Vanilla SDG optimizer (lr=0.01) and Vanilla Adam-Optimizer in Google Colab after installing pysyft with `!pip install syft` and GPU computing enabled. 

**Google Colab:** https://colab.research.google.com/drive/1IIoSdT3oGf8VMaoGjp9_8Ti3q5ILAtDQ

**Expected behavior**
Should have trained to the end as it did when using Vanilla SGD as optimizer.

**Comment by roblewis1237** [comment]
also would like to know the answer to this one. some previous issues suggest fl.optimizer module might have resolved this, but it's unclear to me how to use that?

**Comment by roblewis1237** [comment]
hey @NiWaRe - i looked through the issues and PRs. I think this might resolve it for us 

https://github.com/OpenMined/PySyft/pull/3179/files

there's now an example in https://github.com/OpenMined/PySyft/blob/master/examples/tutorials/Part%2002%20-%20Intro%20to%20Federated%20Learning.ipynb

**Comment by roblewis1237** [comment]
@NiWaRe - I can confirm this works. i can now federate the adam optimizer. the example in the tutorial 2 is accurate. 

thanks @rimijoker + @iamtrask!

**Comment by NiWaRe** [comment]
@roblewis1237 Perfect, thanks for checking in further!
